components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/par/OAuth2ParEndpoint.java

```diff
@@ -0,0 +1,230 @@
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.endpoint.par;
+
+import org.apache.catalina.util.ParameterMap;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.cxf.interceptor.InInterceptors;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.oauth.client.authn.filter.OAuthClientAuthenticatorProxy;
+import org.wso2.carbon.identity.oauth.common.OAuth2ErrorCodes;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth.endpoint.exception.ParErrorDTO;
+import org.wso2.carbon.identity.oauth.endpoint.util.EndpointUtil;
+import org.wso2.carbon.identity.oauth.par.common.ParConstants;
+import org.wso2.carbon.identity.oauth.par.model.ParAuthResponseData;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2ClientValidationResponseDTO;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+/**
+ * REST implementation for OAuth2 PAR endpoint.
+ */
+@Path("/par")
+@InInterceptors(classes = OAuthClientAuthenticatorProxy.class)
+public class OAuth2ParEndpoint {
+
+    private static final Log log = LogFactory.getLog(OAuth2ParEndpoint.class);
+    private final ParHandler parHandler = new ParHandler();
+    private static long scheduledExpiryTime;
+    private static final String ERROR = "error";
+    private static final String ERROR_DESCRIPRION = "error_description";
+
+    @POST
+    @Path("/")
+    @Consumes("application/x-www-form-urlencoded")
+    @Produces("application/json")
+    public Response par(@Context HttpServletRequest request, @Context HttpServletResponse response,
+                        MultivaluedMap<String, String> paramMap) throws ParErrorDTO, Exception {
+
+        setScheduledExpiryTime(Calendar.getInstance(TimeZone.getTimeZone(ParConstants.UTC)).getTimeInMillis());
+
+        OAuth2ClientValidationResponseDTO validationResponse = ParHandler.validateClient(request);
+
+        if (!validationResponse.isValidClient()) {
```

Comment:
Have a private method that does the validations. (Throw error or a boolean.) Error is better